core/src/main/java/org/elasticsearch/common/unit/Fuzziness.java

```diff
@@ -106,9 +115,12 @@ public static Fuzziness parse(XContentParser parser) throws IOException {
                             return build(fuzziness);
                     }
                 } catch (NumberFormatException ex) {
-                    return build(fuzziness);
+                    try {
```

Comment:
This is to properly parse floats into floats and not into strings, so that the query from the parser is the same as from the builder. I would have errors such as "0.6" != 0.6.